spec/README.md

```diff
@@ -0,0 +1,90 @@
+Test Levels
+===========================
+
+This folder contains tests at the unit, acceptance, and integration level for the "puppetlabs-iis" module. The unit 
+tests are kept in the "spec/unit" level folder of the repository. Acceptance in the "spce/acceptance" and Integration at
+the "spec/integration" level.
+
+## Acceptance Folder
+
+At Puppet Labs we define an "acceptance" test as:
+
+> Validating the system state and/or side effects while completing a stated piece of functionality within a tool.
+> This type of test is contained within the boundaries of a tool in order to test a defined functional area within
+> that tool.
+
+What this means for this project is that we will install and configure some infrastructure needed for a "puppetlabs-iis"
+environment. (Puppet agent only.)
+
+## Integration Folder
+
+These tests were originally written by the QA team at Puppet Labs and is actively maintained by the QA team.
+Feel free to contribute tests to this folder as long as they are written with [Beaker](https://github.com/puppetlabs/beaker)
+and follow the guidelines below.
+
+The "puppetlabs-iis" project already contains RSpec and acceptance tests and you might be wondering why there
+is a need to have a set of tests separate from those tests. At Puppet Labs we define an "integration" test as:
+
+> Validating the system state and/or side effects while completing a complete life cycle of user stories using a
+> system. This type of test crosses the boundary of a discrete tool in the process of testing a defined user
+> objective that utilizes a system composed of integrated components.
+
+What this means for this project is that we will install and configure all infrastructure needed in a real-world
+"puppetlabs-iis" environment. (Puppet master and agent.)
+
+## Nodesets Folder
+
+The "acceptance/nodesets" folder contains Beaker host configuration files for the various test platforms used by the 
+"acceptance" test suite. Note: these configs are by default dynamically generated using [Beaker-HostGenerator](https://github.com/puppetlabs/beaker-hostgenerator)
+The default config can be overridden by setting the PLATFORM environment variable.
+
+## Running Tests
+
+### General Setup Steps:
+Option 1 - using Bundler:
+1. Install Bundler
+```
+gem install bundler
+```
+2. Install dependent gems
+```
+bundle install --path .bundle/gems
+```
+3. To see what tasks are available run
+```
+bundle exec rake -T
+```
+
+Option 2 - if not using Bundler, then install the dependent gems.
```

Comment:
We are always recommending bundler usage for community users. People who want to use a different product know their way around anyways. You can point that out here, and continue below with only bundler usage.